tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationRuntimeTest.java

```diff
@@ -221,9 +231,58 @@ public void testConstTypeAnnotAccess() {
         BRunUtil.invoke(resultOne, "testConstTypeAnnotAccess");
     }
 
+    @Test
+    public void testObjectTypeAnnotations() {
+        BRunUtil.invoke(resultOne, "testObjectTypeAnnotations");
+    }
+
+    @Test
+    public void testServiceObjectTypeAnnotations() {
+        BRunUtil.invoke(resultOne, "testServiceObjectTypeAnnotations");
+    }
+
     @AfterClass
     public void tearDown() {
         resultOne = null;
         resultAccessNegative = null;
     }
+
+    public static Object getMethodAnnotations(BTypedesc bTypedesc, BString method, BString annotName) {
+        Type describingType = TypeUtils.getImpliedType(bTypedesc.getDescribingType());
+        int tag = describingType.getTag();
+        assert tag == TypeTags.OBJECT_TYPE_TAG || tag == TypeTags.SERVICE_TAG;
+
```

Comment:
```suggestion
```

Reply:
Added as bit of a split between validation and using as an object type.

Reply:
Since it was intentional, I'm perfectly fine with it.